### PR TITLE
Adds T-42 LMG to PFCs 2.0

### PR DESCRIPTION
**Pull request by Kuro-ai**
## About The Pull Request

Here we go again,DAKKA DAKKA

## Why It's Good For The Game

It's cool

## Changelog
:cl:
add: Added the T-42 LMG for PFCs
/:cl:

### CODE REVIEW

**Comment by tgstation-server** [comment]
You currently have a negative Fix/Feature pull request delta of -5. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

File: code/datums/gamemodes/distress.dm [hunk 1]
```diff
@@ -307,6 +307,7 @@
 						/obj/item/ammo_magazine/rifle/ap = round(scale * 5),
 						/obj/item/ammobox/ext = round(scale * 1),
 						/obj/item/ammo_magazine/rifle/extended = round(scale * 5),
+						/obj/item/weapon/gun/rifle/standard_lmg = round(scale * 15),
```
Comment (factoryman942) [inline code/datums/gamemodes/distress.dm]:
```suggestion
						/obj/item/ammo_magazine/rifle/standard_lmg = round(scale * 15),
```

File: code/modules/projectiles/updated_projectiles/magazines/rifles.dm [hunk 2]
```diff
@@ -92,19 +92,18 @@
...
 
 //-------------------------------------------------------
-//M41AE2 HEAVY PULSE RIFLE
+//T-42 Light Machine Gun
 
 /obj/item/ammo_magazine/rifle/standard_lmg
-	name = "\improper M41AE2 ammo box (10x24mm)"
-	desc = "A semi-rectangular box of rounds for the M41AE2 Heavy Pulse Rifle."
-	icon_state = "m41ae2"
+	name = "\improper T-42 ammo box (10x24mm)"
```
Comment (factoryman942) [inline code/modules/projectiles/updated_projectiles/magazines/rifles.dm]:
```suggestion
	name = "\improper T-42 drum magazine (10x24mm)"
```

File: code/modules/projectiles/updated_projectiles/guns/rifles.dm [hunk 3]
```diff
@@ -345,42 +345,40 @@
...
+//T-42 Light Machine Gun
 
-/obj/item/weapon/gun/rifle/standard_lmg
-	name = "\improper M41AE2 heavy pulse rifle"
-	desc = "A large weapon capable of laying down supressing fire. Currently undergoing field testing. Like it's smaller brother, the M41A1, the M41AE2 is chambered in 10mm."
-	icon_state = "m41ae2"
-	item_state = "m41ae2"
+obj/item/weapon/gun/rifle/standard_lmg
+	name = "\improper T-42 Light Machine Gun"
+	desc = "The T-42 LMG is the TGMCs current standard non-IFF capable LMG. It's known for its ability to put down heavy fire support very well. It is generally used when someone wants to hold a position or provide fire support. It uses 10x24mm"
```
Comment (factoryman942) [inline code/modules/projectiles/updated_projectiles/guns/rifles.dm]:
```suggestion
	desc = "The T-42 LMG is the TGMC's current standard non-IFF-capable LMG. It's known for its ability to lay down heavy fire support very well. It is generally used when someone wants to hold a position or provide fire support. It uses 10x24mm ammunition."
```
also
>It's known for [...] fire support
>It is generally used when someone wants to [...] provide fire support

no shit

File: code/modules/projectiles/updated_projectiles/guns/rifles.dm [hunk 4]
```diff
@@ -345,42 +345,40 @@
...
 //-------------------------------------------------------
-//M41AE2 HEAVY PULSE RIFLE
+//T-42 Light Machine Gun
 
-/obj/item/weapon/gun/rifle/standard_lmg
-	name = "\improper M41AE2 heavy pulse rifle"
-	desc = "A large weapon capable of laying down supressing fire. Currently undergoing field testing. Like it's smaller brother, the M41A1, the M41AE2 is chambered in 10mm."
-	icon_state = "m41ae2"
-	item_state = "m41ae2"
+obj/item/weapon/gun/rifle/standard_lmg
```
Comment (factoryman942) [inline code/modules/projectiles/updated_projectiles/guns/rifles.dm]:
```suggestion
/obj/item/weapon/gun/rifle/standard_lmg
```

File: code/modules/projectiles/updated_projectiles/guns/rifles.dm [hunk 5]
```diff
@@ -345,42 +345,41 @@
 
 
 //-------------------------------------------------------
-//M41AE2 HEAVY PULSE RIFLE
+//T-42 Light Machine Gun
 
 /obj/item/weapon/gun/rifle/standard_lmg
-	name = "\improper M41AE2 heavy pulse rifle"
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/guns/rifles.dm]: Copy this into a new gun, something like `/obj/item/weapon/gun/rifle/m41ae2_hpr`
The repath was to allow for this easily.

File: _maps/map_files/Sulaco/Sulaco.dmm [hunk 6]
```diff
@@ -23268,7 +23268,7 @@
 /obj/item/explosive/grenade/frag,
 /obj/item/explosive/grenade/frag,
 /obj/item/explosive/grenade/frag,
-/obj/item/ammo_magazine/rifle/standard_lmg,
```
Comment (Rohesie) [inline _maps/map_files/Sulaco/Sulaco.dmm]: Leave the old HPR on the Sulaco.

File: code/game/objects/items/storage/belt.dm [hunk 7]
```diff
@@ -414,7 +414,7 @@
 	desc= "A small, lightweight pouch that can be clipped onto Armat Systems M3 Pattern armor or your belt to provide additional storage."
 	storage_slots = 3
 	w_class = WEIGHT_CLASS_BULKY
-	cant_hold = list(/obj/item/ammo_magazine/rifle/standard_lmg)
+	cant_hold = list(/obj/item/ammo_magazine/standard_lmg)
```
Comment (Rohesie) [inline code/game/objects/items/storage/belt.dm]: Is this intended?

Comment (Kuro-ai) [reply code/game/objects/items/storage/belt.dm]: yes,the g8 pouch should't be able to hold it I guess

File: code/game/objects/machinery/vending/new_marine_vendors.dm [hunk 8]
```diff
@@ -969,8 +969,8 @@ GLOBAL_LIST_INIT(available_specialist_sets, list("Scout Set", "Sniper Set", "Dem
 		/obj/item/explosive/grenade/incendiary = list(CAT_LEDSUP, "M40 HIDP incendiary grenade", 3, "black"),
 		/obj/item/explosive/grenade/frag = list(CAT_LEDSUP, "M40 HEDP grenade", 3, "black"),
 		/obj/item/explosive/grenade/impact = list(CAT_LEDSUP, "M40 IMDP grenade", 3, "black"),
-		/obj/item/weapon/gun/rifle/standard_lmg = list(CAT_LEDSUP, "M41AE2 heavy pulse rifle", 12, "orange"),
-		/obj/item/ammo_magazine/rifle/standard_lmg = list(CAT_LEDSUP, "M41AE2 magazine", 4, "black"),
+		/obj/item/weapon/gun/rifle/standard_lmg = list(CAT_LEDSUP, "T-42 LMG", 12, "orange"),
```
Comment (Rohesie) [inline code/game/objects/machinery/vending/new_marine_vendors.dm]: The leader won't need these anymore. They are for PFCs now. Remove both.

**Comment by BurgerLUA** [comment]
Unironically the second worst balanced PR that I've come across. I changed the fire delay to 0.2 seconds as well in my PR and that was only meant for an Alpha Squad leader which contained at most FOUR marines with it. It absolutely fucking shreds xenos who aren't warriors. Literally everyone is using it and it exclusively.

**Comment by Kuro-ai** [comment]
> Unironically the second worst balanced PR that I've come across. I changed the fire delay to 0.2 seconds as well in my PR and that was only meant for an Alpha Squad leader which contained at most FOUR marines with it. It absolutely fucking shreds xenos who aren't warriors. Literally everyone is using it and it exclusively. It's so used, all the starting ammo is gone for it in crash.

This is not the hpr,it has a lot of differences and if you cant bother reading them I won't bother discussing with you

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-7). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Comment by BurgerLUA** [comment]
> > Unironically the second worst balanced PR that I've come across. I changed the fire delay to 0.2 seconds as well in my PR and that was only meant for an Alpha Squad leader which contained at most FOUR marines with it. It absolutely fucking shreds xenos who aren't warriors. Literally everyone is using it and it exclusively. It's so used, all the starting ammo is gone for it in crash.
> 
> This is not the hpr,it has a lot of differences and if you cant bother reading them I won't bother discussing with you

I literally took the the time to look through the code and compare and contrast the stats of this gun and various other guns. I think I was wrong in saying that it was like the HPR, so I change it to "It is a lot like the minigun."

At the time of my comment, it was literally better than a minigun in terms of damage. It was better than anything the marines had and I think it honestly had more damage output and accuracy than a specialist battle rifle. The attachments it got were a meme that allowed it to be as accurate as one, and the damage was on par with the minigun, save for the armor penetration of 15.

I had to convince Roh to give it nerfs, but it looks like most of what I suggested to her was ignored anyways because now for some reason it has 15% less damage and more despite that not being the issue. It's accurate as hell and fires fast.

**Pull request merged**